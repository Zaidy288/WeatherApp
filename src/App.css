* {
  /*This sets a margin anf padding for everything*/
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

body {
  /*This changes the font of the body*/
  font-family: 'montseratt', sans-serif;
}

.app {
  /*This sets the background image for the page*/
  background-image: url('./Images/warm.jpg');
  background-size: cover;
  background-position: bottom;
  transition: 0.4 ease;
}

.app.warm {
  /*Background image*/
  background-image: url('./Images/warm.jpg');
}

main {
  min-height: 100vh;
  background-image: linear-gradient(to bottom, rgba(0, 0, 0, 0.2), rgba(0, 0, 0, 0.75));
  padding: 25px;
}

.search-box {
  /*This adds width and margin to the box*/
  width: 100%;
  margin: 0 0 75px;
}

.search-box .search-bar {
  /*This customizes the display of the search bar*/
  display: block;
  width: 100%;
  padding: 15px;

  appearance: none;
  background: none;
  border: none;
  outline: none;

  background-color: rgba(255, 255, 255, 0.5);
  border-radius: 0px 0px 16px 16px;
  margin-top: -25px;

  box-shadow: 0px 5px rgba(0, 0, 0, 0.2);

  color: #313131;
  font-size: 20px;

  transition: 0.4s ease;
}

.search-box .search-bar:focus {
  /*Changes the way the search bar looks when you are typing*/
  background-color: rgba(255, 255, 255, 0.75);
}

.location-box .location {
  /*The way the location appearance looks*/
  color: #FFF;
  font-size: 32px;
  font-weight: 500;
  text-align: center;
  text-shadow: 3px 3px rgba(50, 50, 70, 0.5);
}

.location-box .date {
  /*The way the date looks*/
  color: #FFF;
  font-size: 20px;
  font-weight: 300;
  font-style: italic;
  text-align: center;
  text-shadow: 2px 2px rgba(50, 50, 70, 0.5);
}

.weather-box {
  /*Centers the box*/
  text-align: center;
}

.weather-box .temp {
  /*The way the temperature shows*/
  position: relative;
  display: inline-block;
  margin: 30px auto;
  background-color: rgba(255, 255, 255, 0.2);
  border-radius: 16px;

  padding: 15px 25px;

  color: #FFF;
  font-size: 102px;
  font-weight: 900;

  text-shadow: 3px 6px rgba(50, 50, 70, 0.5);
  text-align: center;
  box-shadow: 3px 6px rgba(0, 0, 0, 0.2);
}

.weather-box .weather {
  /*The description of the weather*/
  color: #FFF;
  font-size: 48px;
  font-weight: 700;
  text-shadow: 3px 3px rgba(50, 50, 70, 0.5);
}