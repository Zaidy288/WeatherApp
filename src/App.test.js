import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import App from './App';


it('renders correctly', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Fetches the API properly', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
}); 