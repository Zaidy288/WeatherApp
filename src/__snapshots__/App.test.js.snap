// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="app"
>
  <main>
    <div
      className="search-box"
    >
      <input
        className="search-bar"
        onChange={[Function]}
        onKeyPress={[Function]}
        placeholder="Search any city then press enter..."
        type="text"
        value=""
      />
    </div>
    
  </main>
</div>
`;
